theano/sandbox/rng_mrg.py

```diff
@@ -441,8 +444,9 @@ def c_code(self, node, name, inp, out, sub):
 
         for (int i = 0; i < %(ndim)s; ++i)
         {
-            odims[i] = *(dtype_%(size)s *)PyArray_GETPTR1(%(size)s, i);
-            n_elements *= odims[i];
+            odims_i = *(dtype_%(size)s *)PyArray_GETPTR1(%(size)s, i);
+            odims[i] = odims_i;
```

Comment:
If `odims_i` does not fit in an `npy_intp`, then `n_elements` will be greater than `2**31 - 1`, so there will be an error.